File: examples/typescript_variants/shape2/step3/Rectangle.ts
class Rectangle extends Shape {

    height:number
    width:number

    constructor(width:number,height:number) {
        super();
        this.width = width
        this.height = height
    }

    getArea = () => {
        return this.width * this.height
    }

    getName(){
        return "Rectangle"
    }
}

File: examples/dpl2_variants/V4/src/dpl2/Rectangle.java
package dpl2;

/** An axis-aligned rectangle, defined by its top-left corner, width, and height. */
public record Rectangle(Point topLeft, double width, double height) implements Shape {

	public Rectangle {
		if (width <= 0 || height <= 0) {
			throw new IllegalArgumentException("width and height must be positive");
		}
	}

	public double area() {
		return width * height;
	}
}
